tee-worker/omni-executor/intent/asset-lock/src/account_wide.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::sync::RwLock;
+
+use crate::AssetsLock;
+
+// Manages whether any asset is locked for particular account or not.
+pub struct AccountWideAssetsLock {
+	locked: RwLock<bool>,
+}
+
+impl AssetsLock for AccountWideAssetsLock {
+	fn with_lock(
+		_asset_id: crate::AssetId,
+		amount_to_lock: u128,
+		available_amount: u128,
+	) -> Result<Self, ()>
+	where
+		Self: Sized,
+	{
+		if amount_to_lock > available_amount {
+			return Err(());
+		}
+		Ok(Self { locked: RwLock::new(true) })
+	}
+
+	fn lock(
+		&mut self,
+		_asset_id: crate::AssetId,
+		_amount_to_lock: u128,
+		_available_amount: u128,
+	) -> Result<(), ()> {
+		let mut locked = self.locked.write().unwrap();
+
+		if *locked {
+			return Err(());
+		}
+		*locked = true;
+		Ok(())
+	}
+
+	fn release(&mut self, _asset_id: crate::AssetId, _amount_to_release: u128) -> Result<(), ()> {
+		let mut locked = self.locked.write().unwrap();
+
+		if !*locked {
```

Comment:
We could go with less restrictive strategy here and return `Ok` even if not locked.